CryptX/ApplicationLifeCycle/SettingsManager.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  SettingsManager.swift
+//  CryptX
+//
+//  Created by Mehmet Barkın Mercan on 20.07.2024.
+//
+
+import Foundation
+import UIKit
+
+class SettingsManager {
+    static let shared = SettingsManager()
+    
+    fileprivate let userDefaults = UserDefaults.standard
+    
+    private init() {
+        if userDefaults.object(forKey: "settingsArray") == nil {
+            let numbers = (1...100).map { String($0) }
+            let formattedNumbers = numbers.joined(separator: ",")
+            getAndSetCoins(ids: formattedNumbers)
+        }
+        startPeriodicUpdates()
+    }
+    
+    // MARK: - User Defaults Variables
+    
```

Comment:
Instead of using Dictionary, you can create your own struct or class to manage selected and all coins.